nixos/doc/manual/release-notes/rl-2405.section.md

```diff
@@ -52,6 +52,9 @@ The pre-existing [services.ankisyncd](#opt-services.ankisyncd.enable) has been m
   - If [`system.stateVersion`](#opt-system.stateVersion) is >=23.11, `pkgs.nextcloud27` will be installed by default.
   - Please note that an upgrade from v26 (or older) to v28 directly is not possible. Please upgrade to `nextcloud27` (or earlier) first. Nextcloud prohibits skipping major versions while upgrading. You can upgrade by declaring [`services.nextcloud.package = pkgs.nextcloud27;`](options.html#opt-services.nextcloud.package).
 
+- `security.pam.u2f` now follows RFC42.
+  All module options are now settable through the freeform `.settings`.
+
```

Comment:
I don't think this is a backward incompatibility since you added the renamed option modules. It will be a backward incompatibility when those rename modules are eventually removed.

When I add this patch to a config using `u2f`, the build succeeds and just logs these warnings:

```
trace: warning: The option `security.pam.u2f.cue' defined in `/etc/nixos/security/u2f.nix' has been renamed to `security.pam.u2f.settings.cue'.
trace: warning: The option `security.pam.u2f.authFile' defined in `/etc/nixos/security/u2f.nix' has been renamed to `security.pam.u2f.settings.authfile'.
```

Reply:
Arguably the warnings themselves are noteworthy, but I guess this would fit under "other notable changes" just as well.

Reply:
Just for the record:
- all the RFC42 changes for 23.05 were listed under the notable changes section https://nixos.org/manual/nixos/stable/release-notes#sec-release-23.05-notable-changes
- for 23.11, the two mentions of it are under breaking changes
  - `services.knot` added `.settings`, which supersedes `.extraConfig`. The latter was deprecated, but not removed, and will throw a warning if used.
  - `services.sslh` added the appropriate rename modules

Since this isn't a hard breaking change but a deprecation, I'd be inclined to agree that "notable changes" is a better fit.